packages/vhotplug/vhotplug/vhotplug/vhotplug.py

```diff
@@ -0,0 +1,322 @@
+            logger.error("Device not found")
+            return False
+        vm = config.get_vm(new_vm)
+        asyncio.run(remove_usb_device(config, usb_info, None))
+        asyncio.run(attach_usb_device(config, None, usb_info, vm))
+        return True
+    except RuntimeError as e:
+        logger.error("Failed to attach device: %s", e)
+        return False
+
+
+async def async_main():
+    parser = argparse.ArgumentParser(
+        description="Hot-plugging USB devices to the virtual machines"
+    )
+    parser.add_argument(
+        "-c", "--config", type=str, required=True, help="Path to the configuration file"
+    )
+    parser.add_argument(
+        "-a",
+        "--attach-connected",
+        default=False,
+        action=argparse.BooleanOptionalAction,
+        help="Attach connected devices on startup",
+    )
+    parser.add_argument(
+        "-u",
+        "--usb-passthrough-manager",
+        default=True,
+        action=argparse.BooleanOptionalAction,
+        help="With USB passthrough manager",
+    )
+    parser.add_argument(
+        "-t",
+        "--target-cid",
+        type=int,
+        default=5,
+        help="cid of target vm (gui-vm) to send hotplug events to",
+    )
+    parser.add_argument(
+        "-p",
+        "--target-port",
+        type=int,
+        default=7000,
+        help="port of target vm (gui-vm) to send hotplug events to",
+    )
+    parser.add_argument(
+        "-d",
+        "--debug",
+        default=False,
+        action=argparse.BooleanOptionalAction,
+        help="Enable debug messages",
+    )
+    args = parser.parse_args()
+
+    handler = logging.StreamHandler()
+    handler.setFormatter(logging.Formatter("%(levelname)s %(message)s"))
+    logger.addHandler(handler)
+    logger.setLevel(logging.DEBUG if args.debug else logging.INFO)
+
+    if not os.path.exists(args.config):
+        logger.error("Configuration file %s not found", args.config)
+        return
+
+    config = Config(args.config)
+
+    context = pyudev.Context()
+    if args.attach_connected:
+        await attach_connected_devices(context, config)
+
+    monitor = pyudev.Monitor.from_netlink(context)
+
+    watcher = FileWatcher()
+    for vm in config.get_all_vms():
+        vm_socket = vm.get("socket")
+        if vm_socket:
+            watcher.add_file(vm_socket)
+
+    server = None
+    if args.usb_passthrough_manager:
+        from upm.host.service import HostService
+        from upm.logger import setup_logger as upm_logger_setup
+
+        upm_logger_setup("debug" if args.debug else "info")
+        server = HostService(
+            port=args.target_port,
+            cid=args.target_cid,
+            passthrough_handler=handle_user_device_passthrough_request,
+            metadata=(config, userDevices),
+        )
+        server.start()
+    else:
+        if config.api_enabled():
+            server = APIServer(config, context, asyncio.get_event_loop())
+            server.start()
```

Comment:
Again, we shouldn't use any code from the frontend here. The management app should use the API that is already implemented and can be enabled in the config file.